tests/language/completion/mysql_completion/test_smart_completion_public_schema_only.py

```diff
@@ -0,0 +1,367 @@
+import unittest
+from unittest.mock import Mock
+
+import pytest
+from prompt_toolkit.completion import Completion
+from prompt_toolkit.document import Document
+
+from ossdbtoolsservice.language.completion.mysql_completer import MySQLCompleter
+
+metadata = {
+    'users': ['id', 'email', 'first_name', 'last_name'],
+    'orders': ['id', 'ordered_date', 'status'],
+    'select': ['id', 'insert', 'ABC'],
+    'réveillé': ['id', 'insert', 'ABC']
+}
+
+
+class TestSmartCompletion(unittest.TestCase):
+
+    def setUp(self):
+        self.completer = MySQLCompleter(smart_completion=True)
+
+        tables, columns = [], []
+
+        for table, cols in metadata.items():
+            tables.append((table,))
+            columns.extend([(table, col) for col in cols])
+
+        self.completer.set_dbname('test')
+        self.completer.extend_schemata('test')
+        self.completer.extend_relations(tables, kind='tables')
+        self.completer.extend_columns(columns, kind='tables')
+        self.keywords = self.completer.keywords_tree.keys()
+        self.complete_event = Mock()
+
+
+    def test_empty_string_completion(self):
+        text = ''
+        position = 0
+        result = set(
+            self.completer.get_completions(
+                Document(text=text, cursor_position=position),
+                self.complete_event))
+        self.assertSetEqual(result, set(map(lambda completion: Completion(completion, display_meta='keyword'), sorted(self.keywords))))
+
+
+    def test_select_keyword_completion(self):
+        text = 'SEL'
+        position = len('SEL')
+        result = set(self.completer.get_completions(
+            Document(text=text, cursor_position=position),
```

Comment:
Can you add comments to these tests mentioning what is exactly happening. I have seen it at other places, that makes it easier to understand.